plugins/volumeFilter_test.go

```diff
@@ -0,0 +1,439 @@
+	if e != nil {
+		t.Log("could not make market config volume query")
+		return
+	}
+
+	accountIDs := []string{"accountID"}
+	accountConfigVolumeQuery, e := makeDailyVolumeByDateQuery(db, "", "native", "native", accountIDs, []string{})
+	if e != nil {
+		t.Log("could not make account config volume query")
+		return
+	}
+
+	accountMarketConfigVolumeQuery, e := makeDailyVolumeByDateQuery(db, "", "native", "native", accountIDs, marketIDs)
+	if e != nil {
+		t.Log("could not make account and market config volume query")
+		return
+	}
+
+	testCases := []struct {
+		name           string
+		configValue    string
+		exchangeName   string
+		tradingPair    *model.TradingPair
+		assetDisplayFn model.AssetDisplayFn
+		baseAsset      hProtocol.Asset
+		quoteAsset     hProtocol.Asset
+		db             *sql.DB
+		config         *VolumeFilterConfig
+
+		wantError        error
+		wantSubmitFilter SubmitFilter
+	}{
+		// Note: config can be additional market IDs, make sure to test additionalMarketIds and optionalAccountIDs
+		{
+			name:             "failure - display base",
+			configValue:      "",
+			exchangeName:     "",
+			tradingPair:      &model.TradingPair{Base: "FAIL", Quote: ""},
+			assetDisplayFn:   testAssetDisplayFn,
+			baseAsset:        hProtocol.Asset{},
+			quoteAsset:       hProtocol.Asset{},
+			db:               db,
+			config:           &VolumeFilterConfig{},
+			wantError:        fmt.Errorf("could not convert base asset (FAIL) from trading pair via the passed in assetDisplayFn: cannot recognize the asset FAIL"),
+			wantSubmitFilter: nil,
+		},
+		{
+			name:             "failure - display quote",
+			configValue:      "",
+			exchangeName:     "",
+			tradingPair:      &model.TradingPair{Base: "XLM", Quote: "FAIL"},
+			assetDisplayFn:   testAssetDisplayFn,
+			baseAsset:        testNativeAsset,
+			quoteAsset:       hProtocol.Asset{},
+			config:           &VolumeFilterConfig{},
+			wantError:        fmt.Errorf("could not convert quote asset (FAIL) from trading pair via the passed in assetDisplayFn: cannot recognize the asset FAIL"),
+			wantSubmitFilter: nil,
+		},
+		{
+			name:             "failure - query",
+			configValue:      "",
+			exchangeName:     "",
+			tradingPair:      &model.TradingPair{Base: "XLM", Quote: "XLM"},
+			assetDisplayFn:   testAssetDisplayFn,
+			baseAsset:        testNativeAsset,
+			quoteAsset:       testNativeAsset,
+			config:           &VolumeFilterConfig{},
+			db:               nil,
+			wantError:        fmt.Errorf("could not make daily volume by date Query: could not make daily volume by date action: the provided db should be non-nil"),
+			wantSubmitFilter: nil,
+		},
+		{
+			name:           "success - empty config",
+			configValue:    "",
+			exchangeName:   "",
+			tradingPair:    &model.TradingPair{Base: "XLM", Quote: "XLM"},
+			assetDisplayFn: testAssetDisplayFn,
+			baseAsset:      testNativeAsset,
+			quoteAsset:     testNativeAsset,
+			config:         testEmptyVolumeConfig,
```

Comment:
hmm.. I think this test should have failed. maybe you've found a bug.

Reply:
the logic in the factory method should run config.validate() to ensure this test fails.
we will need to create a config.validate() method -- which does the following (and is also tested):
1. tests that exactly one of `SellBaseAssetCapInBaseUnits` or `SellBaseAssetCapInQuoteUnits` is nil.
2. tests that `mode` is one of the two possible values it uses
3. additionalMarketIDs is non-nil
4. optionalAccountIDs is non-nil

Adding this could create conflicts with PR # 2 so can you make a note to add this as part of PR # 2 and then add this test case to ensure it fails on an invalid config?

Reply:
see related test for this -- `TestMakeVolumeFilterConfig`. Maybe it needs a negative version of the test which fails on invalid entries